### PR TITLE
Dependency updates, JDK17

### PR DESCRIPTION
**Pull request by hduerkop**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -11,13 +11,8 @@ jobs:
     strategy:
       matrix:
         java-version: [11, 12, 13, 14, 15, 16, 17]
-        kotlin-version: [1.4.32, 1.5.31, 1.6.0]
+        kotlin-version: [1.6.21, 1.7.0-RC]
```
Comment (Raibaz) [inline .github/workflows/gradle.yml:14]: I'd keep a 1.4.* and a 1.5.* here.

**Comment by Raibaz** [comment]
Looks like the `versionName` property is not available in the android gradle file:

>  /home/runner/work/mockk/mockk/mockk/android/build.gradle.kts:32:9: Unresolved reference: versionName

Can you PTAL?

Thanks!

**Comment by Raibaz** [comment]
All gradle builds fail with 

> Caused by: java.lang.IllegalArgumentException: error: invalid source release: 17

I'm guessing either gradle needs to be upgraded, or JDK 17 is not supported yet.

**Comment by Raibaz** [comment]
Can you please make sure this builds with earlier JDK versions and not just with 17?

Thanks

File: agent/android/build.gradle [hunk 2]
```diff
@@ -59,8 +59,8 @@ android {
     }
 
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 31
+        minSdkVersion 26
```
Comment (AlvaroBrey) [inline agent/android/build.gradle:62]: Was the minSdkVersion bump needed? 26 is oddly high for a library and will break a lot of builds.

**Pull request merged**